docs/core/extensions/httpclient-http3.md

```diff
@@ -0,0 +1,116 @@
+---
+
+# Use HTTP/3 with HttpClient
+
+[HTTP/3](https://quicwg.org/base-drafts/draft-ietf-quic-http.html) is the third and upcoming major version of HTTP. HTTP/3 uses the same semantics as HTTP/1.1 and HTTP/2: the same request methods, status codes, and message fields apply to all versions. The differences are in the underlying transport. Both HTTP/1.1 and HTTP/2 use TCP as their transport. HTTP/3 uses a new transport technology developed alongside HTTP/3 called [QUIC](https://datatracker.ietf.org/doc/html/draft-ietf-quic-transport-34).
+
+HTTP/3 and QUIC have a number of benefits compared to HTTP/1.1 and HTTP/2:
+
+- Faster response time of the first request. QUIC and HTTP/3 negotiates the connection in fewer round-trips between the client and the server. The first request reaches the server faster.
+- Improved experience when there is connection packet loss. HTTP/2 multiplexes multiple requests via one TCP connection. Packet loss on the connection affects all requests. This problem is called "head-of-line blocking". Because QUIC provides native multiplexing, lost packets only impact the requests where data has been lost.
+- Supports transitioning between networks. This feature is useful for mobile devices where it is common to switch between WIFI and cellular networks as a mobile device changes location. Currently HTTP/1.1 and HTTP/2 connections fail with an error when switching networks. An app or web browsers must retry any failed HTTP requests. HTTP/3 allows the app or web browser to seamlessly continue when a network changes. HttpClient and Kestrel do not support network transitions in .NET 6. It may be available in a future release.
+
+> **Important**
+>
+> HTTP/3 is available in .NET 6 as a _preview feature_ because the HTTP/3 specification is not finalized and behavioral or performance issues may exist in HTTP/3 with .NET 6.
+>
+> For more information on preview features, see [**the preview features specification**](https://github.com/dotnet/designs/blob/main/accepted/2021/preview-features/preview-features.md#are-preview-features-supported).
+>
+> Apps configured to take advantage of HTTP/3 should be designed to also support HTTP/1.1 and HTTP/2. If issues are identified in HTTP/3, we recommended disabling HTTP/3 until the issues are resolved in a future release of .NET.
+
+## HTTP/3 requirements
+
+### HttpClient Settings
+
+HTTP/3 support is in preview, and needs to be enabled via a configuration flag which can be set in the project with:
+
+```xml
+<ItemGroup>
+    <RuntimeHostConfigurationOption Include="System.Net.SocketsHttpHandler.Http3Support" Value="true">
+</ItemGroup>
+```
+
+Or using [`AppContext.SetSwitch`](/dotnet/api/system.appcontext.setswitch).
+
+The HTTP version can be configured by setting `HttpRequestMessage.Version` to 3.0. However, because not all routers, firewalls, and proxies properly support HTTP/3, we recommend configuring HTTP/3 together with HTTP/1.1 and HTTP/2. In HttpClient, this can be done by specifying:
```

Comment:
If you set `HttpRequestMessage.Version` to 2.0 and `HttpRequestMessage.VersionPolicy` to `HttpVersionPolicy.RequestVersionOrHigher` then you won't be able to use HTTP/1.1. With this setting, if neither H/2 nor H/3 is available, you'll get an exception.
If you want to be able to support the whole range and also optionally upgrade (via ALPN for H/2 or via Alt-Svc for H/3), you need to start from 1.1.